### PR TITLE
docs: document API v2; add deprecation note; mention previous API version

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
Issues: https://tracker.mender.io/browse/MEN-1987

NOTE: the first commit is more significant
the rest is just about renaming files:
git mv docs/management_api.yml docs/management_api_v2.yml
and git mv docs/management_api_v2.yml docs/management_api.yml
I've divided renaming operation into two commits so just that git will handle this change better, but it didn't help a lot...

### CODE REVIEW

File: docs/management_api.yml [hunk 1]
```diff
@@ -1,11 +1,14 @@
 swagger: '2.0'
 info:
-  version: '1'
+  version: '2'
   title: Device authentication
   description: |
       An API for device authentication handling.
-
-basePath: '/api/management/v1/devauth/'
+      This API replaces fist version of the device authentiaction management API and device admission management API.
```
Comment (apodogrocki) [inline docs/management_api.yml]: first ... authentication ...

File: docs/management_api_v2.yml [hunk 2]
```diff
@@ -4,7 +4,10 @@ info:
   title: Device authentication
   description: |
       An API for device authentication handling.
-
+      This API replaces fist version of the device authentiaction management API and device admission management API.
```
Comment (mchalski) [inline docs/management_api_v2.yml]: authentiaction->authentication

File: docs/management_api.yml (hunk 1 above)
Comment (kjaskiewiczz) [reply docs/management_api.yml]: fixed

File: docs/management_api_v2.yml (hunk 2 above)
Comment (kjaskiewiczz) [reply docs/management_api_v2.yml]: fixed

**Comment by drewmoseley** [comment]
Sorry for the unexpected merge. I read the header and assumed this was the mender-docs repo.  @maciejmrowiec please let me know if I should revert this.

**Comment by maciejmrowiec** [comment]
had plenty of reviews, no worries

**Pull request merged**